### PR TITLE
WIP VML-1 Move back to JDK 11 and Vaadin 23

### PR DESCRIPTION
**Pull request by cristina-attori**
- Moved back to Spring 2.7.14
- Moved back to JDK 11
- Moved back to Vaadin 23.4.0
- Moved org.apache.commons commons-text to 1.10.0
- Replaced some JDK15+ features (textblocks, records, pattern matching on instance of...)

### CODE REVIEW

File: vaadin-maps-leaflet-flow-demo/src/main/java/software/xdev/vaadin/maps/leaflet/flow/DemoView.java [hunk 1]
```diff
@@ -101,7 +102,64 @@ protected void onAttach(final AttachEvent attachEvent)
 		));
 	}
 	
-	record Example(String route, String name, String desc)
-	{
-	}
+	static final class Example
```
Comment (federico-russo) [inline vaadin-maps-leaflet-flow-demo/src/main/java/software/xdev/vaadin/maps/leaflet/flow/DemoView.java:105]: I'd add a todo to remember to transform this into a record when possible, to minimize differences with the original.

File: .gitignore [hunk 2]
```diff
@@ -103,3 +105,5 @@ vite.generated.ts
 .idea/codeStyles/*
 !.idea/codeStyles/codeStyleConfig.xml
 !.idea/codeStyles/Project.xml
+
```
Comment (federico-russo) [inline .gitignore]: Double blank line

File: .idea/codeStyles/Project.xml [hunk 3]
```diff
@@ -96,4 +97,4 @@
       </indentOptions>
     </codeStyleSettings>
   </code_scheme>
-</component>
+</component>
```
Comment (federico-russo) [inline .idea/codeStyles/Project.xml:100]: Removed blank line

File: vaadin-maps-leaflet-flow-demo/src/main/java/software/xdev/vaadin/maps/leaflet/flow/demo/MultiLayerWithPyramidDemo.java [hunk 4]
```diff
@@ -160,34 +165,142 @@ private void createPolygonAndAddToActiveLayerGroup(final Building building)
...
+			String imageUrl,
+			List<DataPoint> dataPoints
+		)
+		{
+			this.name = name;
+			this.url = url;
+			this.imageUrl = imageUrl;
+			this.dataPoints = dataPoints;
+		}
+
```
Comment (federico-russo) [inline vaadin-maps-leaflet-flow-demo/src/main/java/software/xdev/vaadin/maps/leaflet/flow/demo/MultiLayerWithPyramidDemo.java:251]: Many blank lines. But if the project rules do not forbid this, ok.

**Review by federico-russo — CHANGES_REQUESTED** [review]
How sad, removing all this goodness.

File: vaadin-maps-leaflet-flow-demo/src/main/java/software/xdev/vaadin/maps/leaflet/flow/DemoView.java (hunk 1 above)
Comment (cristina-attori) [reply vaadin-maps-leaflet-flow-demo/src/main/java/software/xdev/vaadin/maps/leaflet/flow/DemoView.java:105]: I have created VML 2, that ask to revert all this commit, when we can move to Vaadin24

File: .gitignore (hunk 2 above)
Comment (cristina-attori) [reply .gitignore]: removed

File: vaadin-maps-leaflet-flow-demo/src/main/java/software/xdev/vaadin/maps/leaflet/flow/demo/MultiLayerWithPyramidDemo.java (hunk 4 above)
Comment (cristina-attori) [reply vaadin-maps-leaflet-flow-demo/src/main/java/software/xdev/vaadin/maps/leaflet/flow/demo/MultiLayerWithPyramidDemo.java:251]: Removed

**Pull request merged**